src/dns_cache.rs

```diff
@@ -526,3 +527,11 @@ impl DnsCache {
             .collect()
     }
 }
+
+/// Returns UNIX time in millis
+pub(crate) fn current_time_millis() -> u64 {
```

Comment:
Can't we reuse the definition in dns_parser?

Reply:
On a second thought, I think duplicating is fine since the idea is to the have dns_parser expose only APIs related to parsing.

Reply:
👍  that's exactly why I did that.